glib/src/thread_guard.rs

```diff
@@ -111,4 +115,42 @@ impl<T> Drop for ThreadGuard<T> {
     }
 }
 
+impl<T: Default> Default for ThreadGuard<T> {
+    fn default() -> Self {
+        Self::new(T::default())
+    }
+}
+
 unsafe impl<T> Send for ThreadGuard<T> {}
+
+impl<T: ToValue + StaticType> ToValue for ThreadGuard<T> {
+    fn to_value(&self) -> Value {
+        T::to_value(self.get_ref())
```

Comment:
This is basically taking `ThreadGuard<T>` and converting it to `Value<T>` effectively getting rid of the guard. This can be problematic and shouldn't be done implicitly.

Reply:
That's bad, from what I understand this will _always_ dump the threadguard.

So the returned ThreadGuard we get when calling `obj.property("thread-guard")` is actually a  guard created on the fly, not the one we put in.